Docs/UPDATING.md

```diff
@@ -0,0 +1,59 @@
+* Update `Examples/BazelXCBBuildService/BUILD.bazel` and `Examples/BazelXCBBuildService/Packge.swift` to point at that new module
+* Update `Examples/BazelXCBBuildService/Sources/RequestHandler.swift` to assign the appropriate `RequestPayload` and `ResponsePayload` types to the `BazelXCBBuildServiceRequestPayload` and `BazelXCBBuildServiceResponsePayload` typealiases.
+* Rename the original `XCBBuildService` in the Xcode bundle to `XCBBuildService.original`, just in case you want it again.
+
+### Development loop:
+* Rebuild: `bazel build BazelXCBBuildService`
+* Move it into the proper folder: `My-New-Xcode.app/Contents/SharedFrameworks/XCBuild.framework/PlugIns/XCBBuildService.bundle/Contents/MacOS/BazelXCBBuildService`
+* Run `write_shim.sh` to set up logging into `/tmp/Bazel-trace/`
+* Restart Xcode
+* Do some actions in Xcode, then view the logs, are there any errors?
+* Make a change in the project fixing an error or adding extra logs you deem necessary.
+* Repeat
+
+## Xcode Debugger
+
+To debug `BazelXCBBuildService` while it is being used by Xcode, you will need two versions of Xcode. (e.g. 12.5.0 and 13.0.0)
+
+1. Main version: The version required to build the app.
+1. Debugging version: A different version that can build and attach to `BazelXCBBuildService`. (It can be a newer or older version than the "main" version of Xcode).
+
+To set up `BazelXCBBuildService` for debugging:
+
+1. Remove `BazelXCBBuildService` from the "debugging" Xcode if it is installed. In the "debugging" Xcode, we want to use the official XCBBuildService, not our custom one.
+	1. To see if it is installed, open this folder inside Xcode:
+		```sh
+		open /Applications/Xcode-<debugging-version-number>.app/Contents/SharedFrameworks/XCBuild.framework/PlugIns/XCBBuildService.bundle/Contents/MacOS/
+		```
+	1. If there are three files (`XCBBuildService.original`, `XCBBuildService`, and `BazelXCBBuildService`), `BazelXCBBuildService` is installed in that version of Xcode. To uninstall it, delete `XCBBuildService` and `BazelXCBBuildService`, and rename `XCBBuildService.original` to `XCBBuildService`.
+1. Open `tools/BazelXCBBuildService` in the "debugging" Xcode and build a debug version of BazelXCBBuildService.
+1. Copy the built, debug version of `BazelXCBBuildService` from the DerivedData folder into the "main" version of Xcode:
+	```sh
+	FOLDER=$(ls ~/Library/Developer/Xcode/DerivedData | grep BazelXCBBuildService)
+	cp ~/Library/Developer/Xcode/DerivedData/$FOLDER/Build/Products/Debug/BazelXCBBuildService \
+	/Applications/Xcode-<main-version>.app/Contents/SharedFrameworks/XCBuild.framework/PlugIns/XCBBuildService.bundle/Contents/MacOS/
+	```
+1. In the "debugging" Xcode Go to "Debug" > "Attach to process by PID or Name" and type in "BazelXCBBuildService" if it's not already pre-filled.
```

Comment:
Instead of doing this, you can launch the debugging version with Xcode, using the "wait for launch" setting, and then when it starts in the "debugging" version of Xcode the "main" version will see it and attach.